src/Silk.Data/MediatR/Handlers/TagRequestHandler.cs

```diff
@@ -0,0 +1,166 @@
+                    Aliases = request.OriginalTag is null ? new() : null
+                };
+
+                _db.Tags.Add(tag);
+                await _db.SaveChangesAsync(cancellationToken);
+                
+                return tag;
+            }
+        }
+
+        public class GetHandler : IRequestHandler<TagRequest.Get, Tag?>
+        {
+            private readonly SilkDbContext _db;
+            public GetHandler(SilkDbContext db)
+            {
+                _db = db;
+            }
+            
+            public async Task<Tag?> Handle(TagRequest.Get request, CancellationToken cancellationToken)
+            {
+                Tag? tag = await _db.Tags
+                    .Include(t => t.OriginalTag)
+                    .Include(t => t.Aliases)
+                    .AsSplitQuery()
+                    .FirstOrDefaultAsync(t => 
+                        t.Name.ToLower() == request.Name.ToLower() 
+                        && t.GuildId == request.GuildId, cancellationToken);
+                return tag;
+            }
+        }
+        
+        public class UpdateHandler : IRequestHandler<TagRequest.Update, Tag>
+        {
+            private readonly SilkDbContext _db;
+            public UpdateHandler(SilkDbContext db)
+            {
+                _db = db;
+            }
+            
+            public async Task<Tag> Handle(TagRequest.Update request, CancellationToken cancellationToken)
+            {
+                Tag tag = await _db.Tags.FirstAsync(t => t.Name.ToLower() == request.Name.ToLower() &&
+                                                         t.GuildId == request.GuildId, cancellationToken);
+                tag.Name = request.NewName ?? tag.Name;
+                tag.Uses = request.Uses ?? tag.Uses;
+                tag.Content = request.Content ?? tag.Content;
+                tag.Aliases = request.Aliases ?? tag.Aliases;
+
+                await _db.SaveChangesAsync(cancellationToken);
+                return tag;
+            }
+        }
+
+        public class DeleteHandler : IRequestHandler<TagRequest.Delete>
+        {
+            private readonly SilkDbContext _db;
+            public DeleteHandler(SilkDbContext db)
+            {
+                _db = db;
+            }
+            
+            public async Task<Unit> Handle(TagRequest.Delete request, CancellationToken cancellationToken)
+            {
+                Tag tag = await _db
+                    .Tags
+                    .Include(t => t.OriginalTag)
+                    .Include(t => t.Aliases)
+                    .AsSplitQuery()
+                    .FirstOrDefaultAsync(t =>
+                        t.Name.ToLower() == request.Name.ToLower() &&
+                        t.GuildId == request.GuildId, cancellationToken);
+                
+                Tag[] aliasedTags = await _db
+                    .Tags
+                    .Include(t => t.OriginalTag)
+                    .Include(t => t.Aliases)
```

Comment:
And here you don't need `Aliases`.  I don't think you can have an alias of an alias if I read the earlier logic correctly?  If you can then this isn't going to delete your aliases' aliases.

Reply:
Aliases of aliases do exist and from basic testing they seem to be deleted without issue.